apps/meteor/client/views/omnichannel/departments/DepartmentAgentsTable/DepartmentAgentsTable.tsx

```diff
@@ -0,0 +1,43 @@
+import { useTranslation } from '@rocket.chat/ui-contexts';
+import React from 'react';
+import type { Control, UseFormRegister } from 'react-hook-form';
+import { useWatch, useFieldArray } from 'react-hook-form';
+
+import { GenericTable, GenericTableBody, GenericTableHeader, GenericTableHeaderCell } from '../../../../components/GenericTable';
+import type { FormValues } from '../EditDepartment';
+import AddAgent from './AddAgent';
+import AgentRow from './AgentRow';
+
+type DepartmentAgentsTableProps = {
+	control: Control<FormValues>;
+	register: UseFormRegister<FormValues>;
+};
+
+function DepartmentAgentsTable({ control, register }: DepartmentAgentsTableProps) {
+	const t = useTranslation();
+	const { fields, append, remove } = useFieldArray({ control, name: 'agentList' });
+	const agentList = useWatch({ control, name: 'agentList' });
+
+	return (
+		<>
+			<AddAgent agentList={agentList} data-qa='DepartmentSelect-AgentsTable' onAdd={append} />
+
```

Comment:
Please make sure to pass an `aria-busy` to the generic table (you can use https://github.com/RocketChat/Rocket.Chat/pull/29110/files as reference) to avoid flaky tests on this table

Reply:
I don't think this applies to this specific table.